### ISSUE TITLE
Opening links in Zen AppImage causes 100% CPU usage

### ISSUE THREAD

**Issue by aljustiet** [zen-browser/desktop#2748]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

When I set the default browser as Zen Browser, it sets this desktop file: `userapp-Zen Browser-08XPW2.desktop`, which has this content, and after every restart of the zen browsers, it creates different desktop files.
Here:
```
❯ fd zen
userapp-Zen Browser-08XPW2.desktop
userapp-Zen Browser-5XDIW2.desktop
userapp-Zen Browser-8SKIW2.desktop
userapp-Zen Browser-AKFWW2.desktop
userapp-Zen Browser-NLUTW2.desktop
userapp-Zen Browser-QV8UW2.desktop
userapp-Zen Browser-T9OPW2.desktop
zen-alpha.desktop
zen-browser-AM.desktop
```
I installed zen through [AppMan](https://github.com/ivan-hc/AppMan).
I tried to manually set the default desktop file using this command: `xdg-mime default zen-browser-AM.desktop x-scheme-handler/https`, but when something opens through, it will have 100% CPU usage all the time.
### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.17

### What platform are you seeing the problem on?

Linux

**Comment by LeMoonStar** [comment]
Hey! The build in "set as default browser" in the AppImage is known to be broken, if I remember correctly, this is also the case for normal Firefox.

When installing the Zen AppImage with our AppImage install bash script, you can simply set the Default in your Desktop Environments settings.
I don't know how AppMan handles this. Can you provide me with the `.desktop` file it creates?
Can you also check if "Zen Browser" is an option in your Desktop Environments settings for the default web browser?

**Comment by aljustiet** [comment]
> Hey! The build in "set as default browser" in the AppImage is known to be broken, if I remember correctly, this is also the case for normal Firefox.
> 
> When installing the Zen AppImage with our AppImage install bash script, you can simply set the Default in your Desktop Environments settings. I don't know how AppMan handles this. Can you provide me with the `.desktop` file it creates? Can you also check if "Zen Browser" is an option in your Desktop Environments settings for the default web browser?

[Desktop file AppMan creates](https://rentry.co/pbkwo72q/raw)
I don't use DE, but instead I use Hyprland. I think I should include my system information on every issue I open. I thought that it was unnecessary information since it is the Zen browser's appimage issue, but my expectations weren't realized.
I set it using the xdg-mime cli tool; it works, but the process that it opens uses 1 thread of my CPU.

**Comment by LeMoonStar** [comment]
> [Desktop file AppMan creates](https://rentry.co/pbkwo72q/raw) I don't use DE, but instead I use Hyprland. I think I should include my system information on every issue I open. I thought that it was unnecessary information since it is the Zen browser's appimage issue, but my expectations weren't realized. I set it using the xdg-mime cli tool; it works, but the process that it opens uses 1 thread of my CPU.

The Zen AppImage not being able to set itself as the default is a known issue - see #1797.
However, setting Zen as default usign an external Desktop file, and having Zen use 100% CPU after opening is weird behavior and the main focus of this issue - I recommend changing the title of this issue to reflect this, as otherwise this has to be handled as a duplicate of #1797.

The `.desktop` file seems fine, and is, I believe, Zen's build in .desktop file (see [AppDir/zen.desktop](https://github.com/zen-browser/desktop/blob/31cc0ce7a80c9abcaad3114558916286ba302a21/AppDir/zen.desktop) ) - So it is unlikely to be the cause of the high CPU usage.
Can you try opening Zen manually by doing `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748`?

**Comment by aljustiet** [comment]
> > [Desktop file AppMan creates](https://rentry.co/pbkwo72q/raw) I don't use DE, but instead I use Hyprland. I think I should include my system information on every issue I open. I thought that it was unnecessary information since it is the Zen browser's appimage issue, but my expectations weren't realized. I set it using the xdg-mime cli tool; it works, but the process that it opens uses 1 thread of my CPU.
> 
> The Zen AppImage not being able to set itself as the default is a known issue - see #1797. However, setting Zen as default usign an external Desktop file, and having Zen use 100% CPU after opening is weird behavior and the main focus of this issue - I recommend changing the title of this issue to reflect this, as otherwise this has to be handled as a duplicate of #1797.
> 
> The `.desktop` file seems fine, and is, I believe, Zen's build in .desktop file (see [AppDir/zen.desktop](https://github.com/zen-browser/desktop/blob/31cc0ce7a80c9abcaad3114558916286ba302a21/AppDir/zen.desktop) ) - So it is unlikely to be the cause of the high CPU usage. Can you try opening Zen manually by doing `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748`?

When I run: `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748`, it opens the link and finishes.

**Comment by LeMoonStar** [comment]
> When I run: `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748`, it opens the link and finishes.

So the 100% CPU issue doesn't occur?

**Comment by LeMoonStar** [comment]
Does it still happen when clicking a link and opening Zen that way?

**Comment by aljustiet** [comment]
> > When I run: `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748`, it opens the link and finishes.
> 
> So the 100% CPU issue doesn't occur?

Right.

**Comment by aljustiet** [comment]
> Does it still happen when clicking a link and opening Zen that way?

![image](https://github.com/user-attachments/assets/33719efa-30f5-4226-8cee-76841ee0de61)

**Comment by aljustiet** [comment]
Weird behavior

**Comment by aljustiet** [comment]
```
❯ xdg-mime query default x-scheme-handler/https
zen-browser-AM.desktop
```

**Comment by LeMoonStar** [comment]
Indeed very weird! We still have to confirm if this is a Zen-Browser issue or something entirely different...

Can you please change the Title of this issue to be more descriptive of the actual issue? something like `Opening links in Zen AppImage causes 100% CPU usage`

**Comment by mr-cheffy** [comment]
Could you please try using the portable version? (tar file)

**Comment by LeMoonStar** [comment]
Can you try testing the following things please?

1. 1. Close all instances of zen, make sure no Zen process is running.
    2. Open a link in zen by clicking a link - Report the behaviour.
2. 1. Open one instance of Zen normally
    2. Open a link in zen by clicking a link - Report the behaviour.

additionally, is the behaviour different when using `xdg-open [URL]` instead of clicking a link?

**Comment by aljustiet** [comment]
> Indeed very weird! We still have to confirm if this is a Zen-Browser issue or something entirely different...
> 
> Can you please change the Title of this issue to be more descriptive of the actual issue? something like `Opening links in Zen AppImage causes 100% CPU usage`

I tested it on the firefox appimage, and it doesn't happen there.

**Comment by RayZ3R0** [comment]
Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah

**Comment by aljustiet** [comment]
> Could you please try using the portable version? (tar file)

I tested the AUR package, and the issue doesn't occur there. I assume the AUR package builds Zen from the tar file.

**Comment by RayZ3R0** [comment]
Yeah the AUR does build from tar, use that instead. Appimage has been reported to be a bit broken sometimes

**Comment by LeMoonStar** [comment]
Even if installing another way works, keep in mind this issue is not resolved as AppImage is a official distribution of Zen, the issue could occur for others and should thus be fixed if possible.

@RayZ3R0 @mauro-balades

**Comment by RayZ3R0** [comment]
We should fr remove appimage installer and add DEB RPM and AUR to the linux section

**Comment by aljustiet** [comment]
> Can you try testing the following things please?
> 
>     1. 1. Close all instances of zen, make sure no Zen process is running.
>        2. Open a link in zen by clicking a link - Report the behaviour.
> 
>     2. 1. Open one instance of Zen normally
>        2. Open a link in zen by clicking a link - Report the behaviour.
> 
> 
> additionally, is the behaviour different when using `xdg-open [URL]` instead of clicking a link?

1. Same
2. Same
When I open link using xdg-open, it doesn't utilize all my thread. However, the command itself stays open, and if I terminate it, the browser closes as well.

**Comment by aljustiet** [comment]
> Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah

Right. But, I think that appimages work faster.

**Comment by LeMoonStar** [comment]
> > Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah
> 
> Right. But, I think that appimages work faster.

No, they are not faster

**Comment by LeMoonStar** [comment]
Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue.
Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.

**Comment by aljustiet** [comment]
> > > Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah
> > 
> > 
> > Right. But, I think that appimages work faster.
> 
> No, they are not faster

However, this [video](https://youtu.be/OftD86RgAcc?si=BVti4dztyRLrt6B9&t=427) from TechHut compares different packaging formats, showing that AppImages perform the fastest.

**Comment by LeMoonStar** [comment]
> > > > Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah
> > > 
> > > 
> > > Right. But, I think that appimages work faster.
> > 
> > 
> > No, they are not faster
> 
> However, this [video](https://youtu.be/OftD86RgAcc?si=BVti4dztyRLrt6B9&t=427) from TechHut compares different packaging formats, showing that AppImages perform the fastest.

There is no reason AppImages should be faster than native installs such as with DEB or RPM.
Flatpak and Snaps are the only ones which add Overhead to the process.
AppImage adds overhead to startup because it has to decompress the application.

**Comment by aljustiet** [comment]
> Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue. Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.

No, xdg-open isn't causing the issue. I just meant that when I open a link with xdg-open while the browser isn't running, it stays open, and if I close it, the browser quits too. But if the browser is already running, it works as expected.

**Comment by LeMoonStar** [comment]
> > Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue. Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.
> 
> No, xdg-open isn't causing the issue. I just meant that when I open a link with xdg-open while the browser isn't running, it stays open, and if I close it, the browser quits too. But if the browser is already running, it works as expected.

So when exactly does the 100% CPU issue occur and when does it not? Can you please document that in an easy-to-understand way?

**Comment by aljustiet** [comment]
> > > > > Are you on arch? You can use the AUR, it works perfectly, the appimage is kinda broken, yeah
> > > > 
> > > > 
> > > > Right. But, I think that appimages work faster.
> > > 
> > > 
> > > No, they are not faster
> > 
> > 
> > However, this [video](https://youtu.be/OftD86RgAcc?si=BVti4dztyRLrt6B9&t=427) from TechHut compares different packaging formats, showing that AppImages perform the fastest.
> 
> There is no reason AppImages should be faster than native installs such as with DEB or RPM. Flatpak and Snaps are the only ones which add Overhead to the process. AppImage adds overhead to startup because it has to decompress the application.

I assume the libraries and other components within it are better optimized for running the application, which is why the AppImages performed the best.

**Comment by aljustiet** [comment]
> > > Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue. Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.
> > 
> > 
> > No, xdg-open isn't causing the issue. I just meant that when I open a link with xdg-open while the browser isn't running, it stays open, and if I close it, the browser quits too. But if the browser is already running, it works as expected.
> 
> So when exactly does the 100% CPU issue occur and when does it not? Can you please document that in an easy-to-understand way?

When I click on a link with the mouse.

**Comment by LeMoonStar** [comment]
> > > > Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue. Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.
> > > 
> > > 
> > > No, xdg-open isn't causing the issue. I just meant that when I open a link with xdg-open while the browser isn't running, it stays open, and if I close it, the browser quits too. But if the browser is already running, it works as expected.
> > 
> > 
> > So when exactly does the 100% CPU issue occur and when does it not? Can you please document that in an easy-to-understand way?
> 
> When I click on a link with a mouse.

While Zen is already running? While it is not running?
`xdg-open [url]` behave EXACTLY THE SAME as clicking on a link?

**Comment by aljustiet** [comment]
> > > > > Earlier you stated that running `/home/aljustiet/.local/bin/zen-browser https://github.com/zen-browser/desktop/issues/2748` directly does not cause this issue. Therefor XDG-Open seems to introduce the issue - as if it doesn't just execute the aforementioned command as declared in the .desktop file.
> > > > 
> > > > 
> > > > No, xdg-open isn't causing the issue. I just meant that when I open a link with xdg-open while the browser isn't running, it stays open, and if I close it, the browser quits too. But if the browser is already running, it works as expected.
> > > 
> > > 
> > > So when exactly does the 100% CPU issue occur and when does it not? Can you please document that in an easy-to-understand way?
> > 
> > 
> > When I click on a link with a mouse.
> 
> While Zen is already running? While it is not running? `xdg-open [url]` behave EXACTLY THE SAME as clicking on a link?

Hah, I just noticed that this issue happens only in Electron applications, like Discord and Tutanota Desktop.

**Comment by aljustiet** [comment]
I guess they don't use xdg-open to open files in the default application.

**Comment by aljustiet** [comment]
Do you have any idea why this happens with Electron applications? I tried opening a link using a Qt app, like Telegram, and it doesn't fully utilize my thread when opened in Telegram.

**Comment by aljustiet** [comment]
However, opening it in Firefox's AppImage within the Electron application doesn't fully utilize my thread.

**Comment by aljustiet** [comment]
Now, I have no clue which side this issue is on.

**Comment by LeMoonStar** [comment]
I found this issue (and comment specific to firefox) on a different projects GitHub page:
https://github.com/warpdotdev/Warp/issues/4177#issuecomment-1961504100

... it doesn't seem to be a Zen-Specific issue. But I can't tell the exact cause of this.

**Comment by salamijason** [comment]
User has not replied in a couple weeks. Closing the issue now for the sake of keeping things clean. Please reopen if issue persists.

**Comment by aljustiet** [comment]
> User has not replied in a couple weeks. Closing the issue now for the sake of keeping things clean. Please reopen if issue persists.

It happens only in AppImage format.

**Comment by salamijason** [comment]
@aljustiet it's still occurring?

**Comment by aljustiet** [comment]
> @aljustiet it's still occurring?

Right. 
![image](https://github.com/user-attachments/assets/c3f019cc-9a18-48d4-804b-93e6c82593a2)

**Comment by Samueru-sama** [comment]
@aljustiet Try again and let me know if you still have the issue. (The twilight releases).

The other issue about the multiple `userapp-Zen Browser-***.desktop` was fixed in https://github.com/zen-browser/desktop/pull/6086 Note however that if you use AM, Zen will still create one extra .desktop because `MOZ_APP_LAUNCHER` doesn't work with symlinks, which is what AM normally points to by default while Zen will make the .desktop point to the actual appimage that the symlink points to.

**Comment by aljustiet** [comment]
> [@aljustiet](https://github.com/aljustiet) Try again and let me know if you still have the issue. (The twilight releases).
> 
> The other issue about the multiple `userapp-Zen Browser-***.desktop` was fixed in [#6086](https://github.com/zen-browser/desktop/pull/6086) Note however that if you use AM, Zen will still create one extra .desktop because `MOZ_APP_LAUNCHER` doesn't work with symlinks, which is what AM normally points to by default while Zen will make the .desktop point to the actual appimage that the symlink points to.

Right, everything works fine now, it doesn't consume one thread fully on newer versions.

**Comment by Samueru-sama** [comment]
> Right, everything works fine now, it doesn't consume one thread fully on newer versions.

Good to hear, the issue was the unnecessary usage of `LD_LIBRARY_PATH` in the `AppRun`. 

This affects all firefox browser based appimages, going to backport this to all other projects.
